src/repositories/updater.rs

```diff
@@ -0,0 +1,366 @@
+        )? {
+            return Ok(Some(row.get("id")));
+        }
+        for updater in self.updaters.iter().filter(|u| u.host() == name.host) {
+            let res = match updater.fetch_repository(&name) {
+                Ok(Some(repo)) => self.store_repository(conn, updater.host(), repo),
+                Err(err) => {
+                    warn!("failed to collect `{}` stats: {}", updater.host(), err);
+                    return Ok(None);
+                }
+                _ => continue,
+            };
+            return match res {
+                Ok(repo_id) => Ok(Some(repo_id)),
+                Err(err) => {
+                    warn!("failed to collect `{}` stats: {}", updater.host(), err);
+                    Ok(None)
+                }
+            };
+        }
+        // It means that none of our updaters have a matching host.
+        Ok(None)
+    }
+
+    pub fn update_all_crates(&self) -> Result<()> {
+        let mut conn = self.pool.get()?;
+        for updater in &self.updaters {
+            info!("started updating `{}` repositories stats", updater.host());
+
+            let needs_update = conn
+                .query(
+                    "SELECT host_id
+                     FROM repositories
+                     WHERE host = $1 AND updated_at < NOW() - INTERVAL '1 day';",
+                    &[&updater.host()],
+                )?
+                .into_iter()
+                .map(|row| row.get(0))
+                .collect::<Vec<String>>();
+
+            if needs_update.is_empty() {
+                info!(
+                    "no `{}` repositories stats needed to be updated",
+                    updater.host()
+                );
+                continue;
+            }
+            for chunk in needs_update.chunks(updater.chunk_size()) {
+                let res = updater.fetch_repositories(chunk)?;
+                for node in res.missing {
+                    self.delete_repository(&mut conn, &node, updater.host())?;
+                }
+                for (_, repo) in res.present {
+                    self.store_repository(&mut conn, updater.host(), repo)?;
+                }
+            }
+            info!("finished updating `{}` repositories stats", updater.host());
+        }
+        Ok(())
+    }
+
+    pub fn backfill_repositories(&self) -> Result<()> {
+        let mut conn = self.pool.get()?;
+        for updater in &self.updaters {
+            info!(
+                "started backfilling `{}` repositories stats",
+                updater.host()
+            );
+
+            let needs_backfilling = conn.query(
+                "SELECT releases.id, crates.name, releases.version, releases.repository_url
+                 FROM releases
+                 INNER JOIN crates ON (crates.id = releases.crate_id)
+                 WHERE repository_id IS NULL AND repository_url LIKE $1;",
+                &[&format!("%{}%", updater.host())],
```

Comment:
Are you sure this behaves like you expect? I think it will search for the literal character '%'.

Reply:
I did covered this in my tests, and it works as it should. 

`%` is kept as-is and is not escaped.